packages/js-drive/lib/core/updateSimplifiedMasternodeListFactory.js

```diff
@@ -95,6 +97,12 @@ function updateSimplifiedMasternodeListFactory(
 
       contextLogger.debug(`SML is updated for core heights ${latestRequestedHeight} to ${coreHeight}`);
 
+      await manageMasternodesIdentities(
+        simplifiedMasternodeList,
+        latestRequestedHeight,
+        coreHeight,
+      );
+
       latestRequestedHeight = coreHeight;
     }
```

Comment:
Let's add one more condition and return false if there is no rotation happened. In beginBlock handler you can check if SML rotated or not and if yes, you can call `manageMasternodesIdentities`.